### PR TITLE
Adding a "starting time" feature into Uniform.py

### PR DESCRIPTION
**Pull request by HermanEiTse**
Adding a small feature "starting time" to the uniform generator.

### CODE REVIEW

**Comment by mcolomermolla** [comment]
Comment: in the other generators, I remember the starting time (T0) to be in ms and not in seconds, let's unify this.

**Review by mcolomermolla — COMMENTED** [review]
- I don't understand why this condition would be needed for TimeHist. Both time series and time hist should have a start and end times from which to get the duration.
- I would modify in time series and time hist the argument duration to tmin and tmax

**Comment by tsengj10** [comment]
I agree, let's go with the names "tmin" and "tmax" as they are clearer, since "start" is already a feature of the time objects.  tmin and tmax are also probably more natural arguments, rather than tmin/start and duration.

As for interpretation:  tmin and tamx should specify offsets in seconds, relative to the reference time of the TimeSeries or TimeHist object.

If a part of the region specified by (tmin,tmax) falls outside the range of the time object (TimeSeries or TimeHist), ignore those parts of the region.

If tmin or tmax fall within a bin of a TimeHist, rather than at a bin edge, then fill that bin with the appropriate fraction.

One should try to make sure the generator behaves appropriately for both a TimeSeries and TimeHist.

File: snewpdag/plugins/gen/Uniform.py [hunk 1]
```diff
@@ -16,29 +16,27 @@
 from snewpdag.values import TimeSeries, TimeHist
 
 class Uniform (Node):
-  def __init__(self, field, rate, start = 0,**kwargs):
+  def __init__(self, field, rate, tmin = 0, tmax = 10, **kwargs):
     self.field = field
     self.rate = rate
-    self.start = start
+    self.tmin = tmin
+    self.tmax = tmax
```
Comment (tsengj10) [inline snewpdag/plugins/gen/Uniform.py:23]: If we have tmin and tmax, what is the purpose of duration?

File: snewpdag/plugins/gen/Uniform.py [hunk 2]
```diff
@@ -16,26 +16,27 @@
...
-      dt = self.duration
-      if isinstance(v, TimeHist):
-        if dt <= 0.0 or dt > v.duration:
-          dt = v.duration
-      elif not isinstance(v, TimeSeries):
+      if self.tmax > self.duration:
+        self.tmax = self.duration
+      if self.tmin < 0:
+        self.tmin = 0
+      if not isinstance(v, TimeHist) and not isinstance(v, TimeSeries):
```
Comment (tsengj10) [inline snewpdag/plugins/gen/Uniform.py:34]: I'd move this condition earlier. It's a pre-requisite for doing anything.

File: snewpdag/plugins/gen/Uniform.py [hunk 3]
```diff
@@ -16,26 +16,27 @@
...
     if self.field in data:
       v = data[self.field]
-      dt = self.duration
-      if isinstance(v, TimeHist):
-        if dt <= 0.0 or dt > v.duration:
-          dt = v.duration
-      elif not isinstance(v, TimeSeries):
+      if self.tmax > self.duration:
+        self.tmax = self.duration
+      if self.tmin < 0:
```
Comment (tsengj10) [inline snewpdag/plugins/gen/Uniform.py:32]: tmin can be negative, i.e., events can be generated before the reference time of the TimeSeries/TimeHist.

**Comment by tsengj10** [comment]
In preparation for CM changes, going to merge soon

**Pull request merged**